### PR TITLE
Add failing test case for array spread downgrade

### PR DESCRIPTION
**Pull request by RikudouSage**
Adding test case for https://github.com/rectorphp/rector/issues/6544

### CODE REVIEW

File: rules-tests/DowngradePhp74/Rector/Array_/DowngradeArraySpreadRector/Fixture/with_iterable_pseudotype_array.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/DowngradePhp74/Rector/Array_/DowngradeArraySpreadRector/Fixture/with_iterable_pseudotype_array.php]: File name must have `.inc` suffix, on this case: `with_iterable_pseudotype_array.php.inc`, also, please add namespace `namespace Rector\Tests\DowngradePhp74\Rector\Array_\DowngradeArraySpreadRector\Fixture;`

File: rules-tests/DowngradePhp74/Rector/Array_/DowngradeArraySpreadRector/Fixture/with_iterable_pseudotype_array.php.inc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        $result = [...$result];
+    }
+
+    private function getData(iterable $iterable): iterable
+    {
+        return [...$iterable];
+    }
+}
+?>
+---
```
Comment (samsonasik) [inline rules-tests/DowngradePhp74/Rector/Array_/DowngradeArraySpreadRector/Fixture/with_iterable_pseudotype_array.php.inc]:
```suggestion
-----
```

**Comment by samsonasik** [comment]
@RikudouSage I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/372

**Comment by RikudouSage** [comment]
Thanks for the info and quick implementation!

**Pull request closed without merging**